packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -115,10 +115,128 @@ const deferActions = actions => {
   return deferred
 }
 
-const getLocalReporter = (activity, reporter) =>
-  activity
-    ? { ...reporter, panicOnBuild: activity.panicOnBuild.bind(activity) }
-    : reporter
+/**
+ * Create a local reporter
+ * Used to override reporter methods with activity methods
+ */
+function getLocalReporter({ activity, reporter }) {
+  // If we have an activity, bind panicOnBuild to the activities method to
+  // join them
+  if (activity) {
+    return { ...reporter, panicOnBuild: activity.panicOnBuild.bind(activity) }
+  }
+
+  return reporter
+}
+
+function extendErrorIdWithPluginName(pluginName, errorMeta) {
+  if (typeof errorMeta === `object`) {
+    const id = errorMeta && errorMeta[`id`]
+
+    if (id) {
+      errorMeta[`id`] = `${pluginName}_${id}`
+    }
+  }
+}
+
+function getErrorMapWthPluginName(pluginName, errorMap) {
+  const entries = Object.entries(errorMap)
+
+  return entries.reduce((memo, [key, val]) => {
+    memo[`${pluginName}_${key}`] = val
+
+    return memo
+  }, {})
```

Comment:
Nit: `reduce` tends to confuse people when reading code

```suggestion
  const memo = {}
  const entries = Object.entries(errorMap)
  entries.forEach(([key, val]) => memo[`${pluginName}_${key}`] = val)
  return memo
```